crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let (status, error, payment_response_data) =
+            get_trustpay_response(item.response, item.http_code)?;
+        Ok(Self {
+            status,
+            response: error.map_or_else(|| Ok(payment_response_data), Err),
+            ..item.data
+        })
+    }
+}
+
+pub fn get_trustpay_response(
+    response: TrustpayPaymentsResponse,
+    status_code: u16,
+) -> CustomResult<
+    (
+        enums::AttemptStatus,
+        Option<types::ErrorResponse>,
+        types::PaymentsResponseData,
+    ),
+    errors::ConnectorError,
+> {
+    match (
+        response.instance_id,
+        response.payment_status,
+        response.payment_request_id,
+        response.gateway_url,
+        response.payment_result_info,
+        response
+            .payment_information_response
+            .clone()
+            .unwrap_or_default()
+            .status,
+    ) {
+        (Some(instance_id), Some(payment_status), _, _, _, _) => {
+            let (status, msg) =
+                get_transaction_status(payment_status.as_str(), response.redirect_url.clone())?;
+            let form_fields = response
+                .redirect_params
+                .unwrap_or(std::collections::HashMap::new());
+            let redirection_data = response.redirect_url.map(|url| services::RedirectForm {
+                endpoint: url,
+                method: services::Method::Post,
+                form_fields,
+            });
+            let error = if msg.is_some() {
+                Some(types::ErrorResponse {
+                    code: payment_status,
+                    message: msg.unwrap_or_default(),
+                    reason: None,
+                    status_code,
+                })
+            } else {
+                None
+            };
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(instance_id),
+                redirection_data,
+                mandate_reference: None,
+                connector_metadata: None,
+            };
+            Ok((status, error, payment_response_data))
+        }
+        (_, _, Some(payment_request_id), Some(gateway_url), _, _) => {
+            let status = enums::AttemptStatus::AuthenticationPending;
+            let error = None;
+            let form_fields: HashMap<String, String> = HashMap::from_iter(
+                gateway_url
+                    .query_pairs()
+                    .map(|(key, value)| (key.to_string(), value.to_string())),
+            );
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    payment_request_id.to_string(),
+                ),
+                redirection_data: Some(services::RedirectForm {
```

Comment:
try to use `services::RedirectForm::from(response.redirect_url, services::Method::Post)`, this might solve your use case